### DISCUSSION TITLE
MacOS implementation for netif mem disk plugins

### DISCUSSION THREAD

**Discussion by mp3monster** [fluent/fluent-bit#8121] [Ideas]
While Mac OS may not be a typical production deployment environment, not having the core infrastructure measurements for network, memory, disk, and CPU means that you can't test basic configurations on MacOS. As these inputs are also good sources to master transforming old school inputs to OTel metrics it would represent an impediment to learning to use these features.

**Comment by patrick-stephens** [comment]
@cosmo0920 would node exporter work for macOS on these?

**Reply by cosmo0920** [reply]
At the current master, it will start to handle collecting mertics on macOS.
And yeah, the current implementation is able to handle: cpu,loadavg,meminfo,diskstats,uname,netdev(netif metrics).
